pkg/cache/cache.go

```diff
@@ -65,8 +65,19 @@ type Response struct {
 	// Proxy responds with this version as an acknowledgement.
 	Version string
 
+	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
+	ResourceMarshaled bool
+
 	// Resources to be included in the response.
 	Resources []Resource
+
+	// Marshaled Resources to be included in the response.
+	MarshaledResources []MarshaledResource
```

Comment:
Maybe we should use `*any.Any` instead of creating a new struct? It fulfills the same purpose, and it's a bit easier to respond after caching serialized resource.

Reply:
Just realized that there is no need to add `ResourceType` into the `MarshaledResources`, and declare it as the binary array. It looks better now.